Src/Newtonsoft.Json/JsonTextWriter.Async.cs

```diff
@@ -317,18 +325,28 @@ public override Task WritePropertyNameAsync(string name, CancellationToken cance
         internal Task DoWritePropertyNameAsync(string name, CancellationToken cancellationToken)
         {
             Task task = InternalWritePropertyNameAsync(name, cancellationToken);
-            if (!task.IsCompleted)
+            switch (task.Status)
             {
-                return DoWritePropertyNameAsync(task, name, cancellationToken);
+                case TaskStatus.Canceled:
+                case TaskStatus.Faulted:
+                    return task;
+                case TaskStatus.RanToCompletion:
+                    break;
```

Comment:
Add a comment that the task was completed in sync and to continue executing on sync path